shared/state.rs

```diff
@@ -301,4 +304,31 @@ impl State {
 
         Ok(())
     }
+
+    /// Checkek that any Queued tasks dont have any failed dependencies, otherwise marked it as failed
+    pub fn update_dependencies(&mut self) {
```

Comment:
`update_dependencies` is probably a bad name for this function. Something like `check_{for?}_failed_dependencies` is probably better.